### ISSUE TITLE
Export core functions of library for reuse

### ISSUE THREAD

**Issue by ChrisLFieldsII** [includable/react-native-map-link#228]
This is the best lib for linking to map apps from my research as it supports the most apps.

The out-of-the-box functionality prob works for 90% of users but I had a use case that this lib didnt cover out-of-the-box for me.

What I needed was the core logic of this lib, determining which map apps are available on a users phone so that I could use this data to build my use cases UI. 

(the use case is to display map app options in an actionsheet with a copy to clipboard option for the address in case user has no map apps or just wants to copy)
<img width="299" alt="Screen Shot 2022-07-17 at 8 52 56 PM" src="https://user-images.githubusercontent.com/30332608/179434900-1a527d82-0511-4525-8b29-7c5215e7cfdc.png">

I created a patch to support my use case. I'll drop the patch in here.

The gist was to export the `generatePrefixes` and `getAvailableApps` functions

```patch
diff --git a/node_modules/react-native-map-link/index.d.ts b/node_modules/react-native-map-link/index.d.ts
index aa09a07..6a239aa 100644
--- a/node_modules/react-native-map-link/index.d.ts
+++ b/node_modules/react-native-map-link/index.d.ts
@@ -49,7 +49,12 @@ interface PopupProps {
   appTitles?: {[key: string]: string};
 }
 
+type MapAppPrefix = 'apple-maps' | 'google-maps' | 'citymapper' | 'uber' | 'lyft' | 'transit' | 'truckmap' | 'waze' | 'yandex' | 'moovit' | 'yandex-maps' | 'yandex-taxi' | 'kakaomap' | 'mapycz' | 'maps-me' | 'osmand' | 'gett' | 'navermap' | 'dgis' | 'liftago' | 'petalmaps'
+type Prefixes = Record<MapAppPrefix, string>
+
 export function showLocation(
   options: Options,
 ): Promise<string | undefined | null>;
 export class Popup extends React.Component<PopupProps> {}
+export function generatePrefixes(opts?: Options): Prefixes
+export function getAvailableApps(prefixes: Prefixes): Promise<MapAppPrefix[]>
diff --git a/node_modules/react-native-map-link/index.js b/node_modules/react-native-map-link/index.js
index 32e3c5d..2d8daf4 100644
--- a/node_modules/react-native-map-link/index.js
+++ b/node_modules/react-native-map-link/index.js
@@ -3,9 +3,11 @@
  */
 
 import Popup from './src/components/Popup'
-import { showLocation } from './src'
+import { showLocation, getAvailableApps, generatePrefixes } from './src'
 
 export {
   showLocation,
-  Popup
+  Popup,
+  getAvailableApps,
+  generatePrefixes
 }
diff --git a/node_modules/react-native-map-link/src/constants.js b/node_modules/react-native-map-link/src/constants.js
index a74291f..26c6dbc 100644
--- a/node_modules/react-native-map-link/src/constants.js
+++ b/node_modules/react-native-map-link/src/constants.js
@@ -6,7 +6,7 @@ import {Platform} from 'react-native';
 
 export const isIOS = Platform.OS === 'ios';
 
-export function generatePrefixes(options) {
+export function generatePrefixes(options={}) {
   return {
     'apple-maps': isIOS ? 'maps://' : 'applemaps://',
     'google-maps': prefixForGoogleMaps(options.alwaysIncludeGoogle),
diff --git a/node_modules/react-native-map-link/src/index.js b/node_modules/react-native-map-link/src/index.js
index d06232d..8e11377 100644
--- a/node_modules/react-native-map-link/src/index.js
+++ b/node_modules/react-native-map-link/src/index.js
@@ -5,7 +5,10 @@
 import {Linking} from 'react-native';
 
 import {generatePrefixes, generateTitles, isIOS} from './constants';
-import {askAppChoice, checkOptions} from './utils';
+import {askAppChoice, checkOptions,} from './utils';
+
+export {getAvailableApps} from './utils'
+export { generatePrefixes }
 
 /**
  * Open a maps app, or let the user choose what app to open, with the given location.
```

**Comment by tschoffelen** [comment]
Hi Chris, thanks for your message. If you open a PR for this change, I'm happy to accept it.

**Comment by ChrisLFieldsII** [comment]
Yes, I'd be happy do that. I will try to open up a PR within a week or so :)

**Comment by tschoffelen** [comment]
Fixed in #232
